R/DataFrameList-class.R

```diff
@@ -164,8 +164,23 @@ setMethod("ROWNAMES", "DataFrameList", function(x) names(x))
 
 setGeneric("commonColnames", function(x) standardGeneric("commonColnames"))
 
-setMethod("commonColnames", "SplitDataFrameList",
-          function(x) colnames(head(x, 1L))[[1L]])
+setMethod("commonColnames", "SimpleSplitDataFrameList",
+          function(x) {
+            if (length(x)) 
+              colnames(x[[1]])
+            else NULL
+          })
+
+setGeneric("commonColnames<-", function(x, value) standardGeneric("commonColnames<-"))
+
+setReplaceMethod("commonColnames", "SimpleSplitDataFrameList", 
```

Comment:
Could we just use the `colnames<-()` methods for this? It looks like they already handle the character vector case, setting it on each element, even for the non-split case.

Reply:
- Kind of confusing, though, when `colnames()` returns a `CharacterList`.
- There's a nice symmetry with `commonColnames()` and `commonColnames()<-`.

Reply:
Sorry, I meant that this could delegate to `colnames<-()` instead of reimplementing it.

Reply:
If you changed the signature to SplitDataFrameList, you could get rid of the CompressedSplitDataFrameList method.